### PR TITLE
Pipelinerun actions

### PR DESCRIPTION
**Pull request by mfrances17**
## Fixes 
Fixes https://issues.redhat.com/browse/HACBS-990

## Description
Stop and delete commands added. Rerun is still TBD.

## Type of change

- [x] Feature

## How to test or reproduce?
Go to the pipelinerun list page, select the kabob on a particular pipelinerun row and select Stop (to cancel a running PLR) or Delete (to bring up a modal and eventually delete the PLR).

## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by christianvogt** [comment]
@mfrances17 I wasn't aware that there was supposed to be a delete action on a pipeline run. Could you share a link to this.

File: src/hacbs/components/Modals/resource-modals.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { PipelineRunKind } from '../../types';
+
+export const pipelineRunDeleteModal = (pipelineRun: PipelineRunKind) =>
+  createDeleteModalLauncher(pipelineRun.kind)({
+    obj: pipelineRun,
+    model: PipelineRunModel,
+    description: (
+      <>
+        The pipelinerun <strong>{pipelineRun.metadata.name}</strong> will be deleted from the
+        pipelinerun view and all the environments it is currently deployed.
```
Comment (christianvogt) [inline src/hacbs/components/Modals/resource-modals.tsx:13]: @mfrances17 I don't think we want to delete pipeline runs. Nor do I believe that deletion of a pipeline run would have such cascading side effects as to remove a deployed snapshot from an environment.

Comment (christianvogt) [reply src/hacbs/components/Modals/resource-modals.tsx:13]: The attachment you referred to was my screen grab from OpenShift Console where I was inquiring about the cancel action.

That's fine to log a new story for rerun and do it later.

Comment (mfrances17) [reply src/hacbs/components/Modals/resource-modals.tsx]: i've removed Delete and will create a new story for Rerun.

File: src/hacbs/components/PipelineRunListView/pipelinerun-actions.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      model: PipelineRunModel,
+      queryOptions: {
+        name: pipelineRun.metadata.name,
+        ns: pipelineRun.metadata.namespace,
+      },
+      patches: [
+        {
+          op: 'replace',
+          path: `/spec/status`,
+          value: 'PipelineRunCancelled',
```
Comment (christianvogt) [inline src/hacbs/components/PipelineRunListView/pipelinerun-actions.tsx:18]: dev console uses `StoppedRunFinally`:
https://github.com/openshift/console/blob/master/frontend/packages/pipelines-plugin/src/utils/pipeline-actions.tsx#L180-L211

File: src/hacbs/components/PipelineRunListView/pipelinerun-actions.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        },
+      ],
+    });
+  };
+
+  return [
+    {
+      cta: () => pipelineRunStop(),
```
Comment (christianvogt) [inline src/hacbs/components/PipelineRunListView/pipelinerun-actions.tsx:26]: You're going to want to disable this action for any pipeline that isn't currently running. 
See https://github.com/openshift/console/blob/master/frontend/packages/pipelines-plugin/src/utils/pipeline-actions.tsx#L180-L211

File: src/hacbs/types/pipelineRun.ts [hunk 4]
```diff
@@ -142,7 +142,7 @@ export type PipelineRunStatus = {
 
 export type PipelineRunKind = K8sResourceCommon & {
   spec: {
-    pipelineRef?: { name: string; bundle?: string };
+    pipelineRef?: { name: string };
```
Comment (karthikjeeyar) [inline src/hacbs/types/pipelineRun.ts:145]: Please revert this change as `Tekton` now supports [bundle](https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#tekton-bundles) field to support remotely defined pipelines.
```suggestion
    pipelineRef?: { name: string, bundle?: string };
```

File: src/hacbs/components/PipelineRunListView/pipelinerun-actions.tsx (hunk 2 above)
Comment (karthikjeeyar) [reply src/hacbs/components/PipelineRunListView/pipelinerun-actions.tsx:18]: Yes, `PipelineRunCancelled` is deprecated.

Comment (mfrances17) [reply src/hacbs/components/PipelineRunListView/pipelinerun-actions.tsx]: thanks, updated to use `StoppedRunFinally`

File: src/hacbs/components/PipelineRunListView/pipelinerun-actions.tsx (hunk 3 above)
Comment (mfrances17) [reply src/hacbs/components/PipelineRunListView/pipelinerun-actions.tsx:27]: changed it so it is only enabled when the PLR is running, thanks

File: src/hacbs/types/pipelineRun.ts (hunk 4 above)
Comment (mfrances17) [reply src/hacbs/types/pipelineRun.ts]: done, thanks!

**Comment by christianvogt** [comment]
/approve

Talk with Karthik as it seems for prekcp we are running an old version of pipelines and the status I told you to use doesn't work here :(
